File: hrpay/api/job_position.py
import frappe
import json

@frappe.whitelist()
def get_job_positions():
    try:
        job_positions = frappe.get_all("Job Position", fields=["name", "job_position"])
        return json.dumps(job_positions)
    except Exception as e:
        frappe.log_error(f"Error fetching job positions: {str(e)}")
        return json.dumps({"error": str(e)})

File: hrpay/api/job_position.js
export async function fetchJobPositions() {
  try {
    const response = await fetch("/api/method/htpay.api.get_job_positions");
    const data = await response.json();
    return data.map(job => ({ id: job.name, name: job.job_position }));
  } catch (error) {
    console.error("Error fetching job positions:", error);
    return [];
  }
}